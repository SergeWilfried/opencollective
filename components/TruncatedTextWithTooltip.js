import React from 'react';
import PropTypes from 'prop-types';
import { truncate } from 'lodash';

import StyledTooltip from './StyledTooltip';
import { Span } from './Text';

/**
 * A tooltip that truncates a value if it's longer than the
 * provided length.
 */
const TruncatedTextWithTooltip = ({ value, cursor, length = 30 }) => {
  if (value?.length <= length) {
    return value;
  } else {
    return (
      <React.Fragment>
        <StyledTooltip containerCursor={cursor} content={() => <Span color="black.100">{value}</Span>}>
          {truncate(value, { length })}
        </StyledTooltip>
      </React.Fragment>
    );
  }
};

TruncatedTextWithTooltip.propTypes = {
  value: PropTypes.string.isRequired,
  length: PropTypes.number,
  cursor: PropTypes.string,
};

export default TruncatedTextWithTooltip;
