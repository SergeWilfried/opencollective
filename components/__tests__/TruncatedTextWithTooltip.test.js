import 'jest-styled-components';

import React from 'react';
import { ThemeProvider } from 'styled-components';

import theme from '../../lib/theme';
import { snapshot } from '../../test/snapshot-helpers';

import TruncatedTextWithTooltip from '../TruncatedTextWithTooltip';

describe('TruncatedTextWithTooltip component', () => {
  it('renders default options', () => {
    snapshot(
      <ThemeProvider theme={theme}>
        <TruncatedTextWithTooltip value="A short string" />
      </ThemeProvider>,
    );
  });

  it('renders default options', () => {
    snapshot(
      <ThemeProvider theme={theme}>
        <TruncatedTextWithTooltip value="a string that is more than 30 characters long" />
      </ThemeProvider>,
    );
  });
});
