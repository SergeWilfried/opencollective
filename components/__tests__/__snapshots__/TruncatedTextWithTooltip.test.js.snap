// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TruncatedTextWithTooltip component renders default options 1`] = `"A short string"`;

exports[`TruncatedTextWithTooltip component renders default options 2`] = `
.c0 {
  display: inline-block;
  cursor: help;
}

.c0 button:disabled {
  pointer-events: none;
}

<div
  className="c0"
  cursor="help"
  data-cy="tooltip-trigger"
  display="inline-block"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  a string that is more than ...
</div>
`;
